### ISSUE TITLE
Google Cloud Storage NIO Checkpointer throws path exception

### ISSUE THREAD

**Issue by coadan** [xtdb/crux#1657]
Maybe there is a way to get it working, but then the documentation is incorrect.

I couldn't get a workable string combining different ways of constructing the string:

`(Path/of (URI. <path>))` 
`(io/file <path>)` <- Just created a local file repository so it probably should throw an exception?
and just the plain string as in the below test config:
```clojure
 :xtdb/index-store    {:kv-store {:xtdb/module  `rocks/->kv-store
                                      :db-dir       (io/file "/tmp/rocksdb/idx")
                                      :checkpointer {:xtdb/module      `check/->checkpointer
                                                     :store            {:xtdb/module `gcs/->checkpoint-store
                                                                        :path        "gs://xtdb-dev-snapshots/index"}
                                                     :approx-frequency (t/duration (t/minutes 2))}}}}
```

Exception:
```
09:46:39.051 [xtdb-checkpoint-1] INFO  xtdb.checkpoint - Uploading checkpoint at '{:xtdb.api/tx-time #inst "2021-11-05T14:13:52.528-00:00", :xtdb.api/tx-id 156}'
09:46:39.059 [xtdb-checkpoint-1] WARN  xtdb.checkpoint - Checkpointing failed
java.nio.file.ProviderMismatchException: Not a Cloud Storage path:  (UnixPath)
	at com.google.cloud.storage.contrib.nio.CloudStorageUtil.checkPath(CloudStorageUtil.java:48)
	at com.google.cloud.storage.contrib.nio.CloudStoragePath.resolve(CloudStoragePath.java:198)
	at com.google.cloud.storage.contrib.nio.CloudStoragePath.resolve(CloudStoragePath.java:48)
	at xtdb.checkpoint$sync_path.invokeStatic(checkpoint.clj:118)
	at xtdb.checkpoint$sync_path.invoke(checkpoint.clj:114)
	at xtdb.checkpoint.FileSystemCheckpointStore.upload_checkpoint(checkpoint.clj:159)
	at xtdb.checkpoint$checkpoint.invokeStatic(checkpoint.clj:43)
	at xtdb.checkpoint$checkpoint.invoke(checkpoint.clj:37)
	at xtdb.checkpoint.ScheduledCheckpointer$run__43530$fn__43534.invoke(checkpoint.clj:70)
	at xtdb.checkpoint.ScheduledCheckpointer$run__43530.invoke(checkpoint.clj:69)
	at xtdb.checkpoint.ScheduledCheckpointer$schedule__43541$fn__43542.invoke(checkpoint.clj:90)
	at clojure.lang.AFn.run(AFn.java:22)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
```

**Comment by alexisvincent** [comment]
We're seeing the same thing. @refset can you confirm the status on this?

**Comment by refset** [comment]
Hey @coadan @alexisvincent - thank you both for reporting this! The fix will be included in the upcoming `1.21.0` release :pray:

**Comment by alexisvincent** [comment]
Thanks @refset 🤍
